src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -730,7 +730,7 @@ private static int WriteHeader(MessageOpcode opcode, byte[] sendBuffer, ReadOnly
 
                     // If this a text message, validate that it contains valid UTF8.
                     if (header.Opcode == MessageOpcode.Text &&
-                        !TryValidateUtf8(payloadBuffer.Span.Slice(0, totalBytesReceived), header.Fin, _utf8TextState))
+                        !TryValidateUtf8(payloadBuffer.Span.Slice(0, totalBytesReceived), header.Fin && header.PayloadLength == 0, _utf8TextState))
```

Comment:
I see `header.Fin && header.PayloadLength == 0` is double computed (https://github.com/dotnet/corefx/pull/29840/files#diff-ffa92551950bc49862828df5ae6518f4R742). Maybe store it in a local called endOfMessage?